pkg/fileutils/reflink_linux.go

```diff
@@ -0,0 +1,20 @@
+package fileutils
+
+import (
+	"io"
+	"os"
+
+	"golang.org/x/sys/unix"
+)
+
+// LinkOrCopy attempts to reflink the source to the destination fd.
+// If reflinking fails or is unsupported, it falls back to io.Copy().
+func LinkOrCopy(src, dst *os.File) error {
+	err := unix.IoctlFileClone(int(dst.Fd()), int(src.Fd()))
+	if err == nil {
+		return nil
+	}
+
+	_, err = io.Copy(dst, src)
```

Comment:
could we use Copy from pkg/pools?  Would the new dependency be a problem?

Reply:
Mhh does this help us here? Would this one allocation for the buffer from io.Copy() even be noticeable. Also io.Copy() uses interfaces such as WriteTo() which if you pass a file should already try to use sendfile() and not need a buffer

Thus I am not sure if using a pre existing buffer would help us here. Also would there not be contention if many goroutines try to use pkg/pool at the same time? 

The more practical concern `pkg/ioutils` is imported from pkg/pools so we cannot import it here and would need to move the function into pkg/pool.

Reply:
Not worth the effort. Thanks